AutomationHelpers/src/UserCodeCollections/EmailLibrary.cs

```diff
@@ -43,41 +43,74 @@ public static class EmailLibrary
             string username = "",
             string password = "")
         {
-            try
-            {
-                var client = new SmtpClient(serverHostname, serverPort)
-                {
-                    Credentials = new NetworkCredential(username, password),
-                    EnableSsl = useSSL,
-                };
+            EmailLibrary.SendEmail(subject, to, from, body, new string[] { attachment }, serverHostname, serverPort, useSSL, username, password);
+        }
 
-                var message = new MailMessage(from, to)
+        /// <summary>
+        /// Sends an email.
+        /// </summary>
+        /// <param name="subject">Email subject</param>
+        /// <param name="to">Email recipient</param>
+        /// <param name="from">Email sender</param>
+        /// <param name="body">Email message</param>
+        /// <param name="attachment">Path of a file to attach</param>
+        /// <param name="serverHostname">Server hostname</param>
+        /// <param name="serverPort">Server port</param>
+        /// <param name="useSSL">Defines whether SSL is used or not (true or false)</param>
+        /// <param name="username">Username</param>
+        /// <param name="password">Password</param>
+        [UserCodeMethod]
+            public static void SendEmail(
```

Comment:
Either pass on exceptions or change the return type to bool and let the caller know if sending succeeded or not.
IMHO passing on/throwing exception would be better - although it would be a breaking change now. But rather now than later...

Reply:
The Email Library now returns a bool, but because the SmtpCLient does return void, it is an assumption. If it fails, there should be an exception which triggers an error log (failed test case).